backend/internal/redmine/api.go

```diff
@@ -60,93 +59,85 @@ type TimeEntry struct {
 	User     int    `json:"user_id"`
 }
 
-func Login(authHeader string, redmineConf cfg.RedmineConfig) bool {
-
-	url := redmineConf.Host + ":" + redmineConf.Port + "/issues.json"
-	method := "GET"
+func doRequest(
+	redmineConf cfg.RedmineConfig,
+	method string, endpoint string,
+	headers map[string]string) (*http.Response, error) {
 
-	var r IssuesRes
+	url := redmineConf.Host + ":" + redmineConf.Port + endpoint
 
-	client := &http.Client{}
 	req, err := http.NewRequest(method, url, nil)
-
 	if err != nil {
-		fmt.Println(err)
-		return false
+		return nil, err
 	}
 
-	req.Header.Add("Authorization", authHeader)
-	req.Header.Add("Content-Type", "application/json")
+	for key, value := range headers {
+		req.Header.Set(key, value)
+	}
+	req.Header.Set("Content-Type", "application/json")
 
+	client := &http.Client{}
 	res, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	return res, nil
+}
```

Comment:
It feels like this function should no longer belong here, as it's very generic

Reply:
If you make host and port as arguments then we could move it to a http module

Reply:
Do you mean to extend the standard `http` class? I'm not sure how to do that. It's generic, but only in the sense that all the other functions in this module would probably use it. In a sense, the `redmine/api` module could be seen as an extension (specialization) of `http`, I suppose, for Redmine. Hmm...

If you can come up with some way of formatting the headers (the string map) in the calls to `doRequest` in a sensible way, that would be nice. It's currently a bit jarring with the way the lines are uneven.

Reply:
I think we can continue to refactor in a future PR. So I am merging since the PR is approved.